src/middleware/validateApiKey.js

```diff
@@ -1,58 +1,19 @@
-import { compareSync } from 'bcrypt';
-import { developerSchema } from '../models/Developer';
 import {
   MAIN_KEY,
-  isTest,
   isDevelopment,
   isProduction,
 } from '../config';
-import { createDbConnection } from '../services/database';
-
-const PROD_LIMIT = 2500;
-const FALLBACK_API_KEY = 'fallback_api_key';
-
-const determineLimit = (apiLimit) => (
-  isTest
-    ? apiLimit || PROD_LIMIT
-    : PROD_LIMIT
-);
-
-const isSameDate = (first, second) => (
-  first.getFullYear() === second.getFullYear()
-    && first.getMonth() === second.getMonth()
-    && first.getDate() === second.getDate()
-);
-
-/* Increments usage count and updates usage date */
-const handleDeveloperUsage = async (developer) => {
-  const updatedDeveloper = developer;
-  const isNewDay = !isSameDate(updatedDeveloper.usage.date, new Date());
-  updatedDeveloper.usage.date = Date.now();
-
-  if (isNewDay) {
-    updatedDeveloper.usage.count = 0;
-  } else {
-    updatedDeveloper.usage.count += 1;
-  }
-
-  return updatedDeveloper.save();
-};
-
-/* Finds a developer with provided information */
-const findDeveloper = async (apiKey) => {
-  console.time('Finding developer account');
-  const connection = createDbConnection();
-  const Developer = connection.model('Developer', developerSchema);
-  const developers = await Developer.find({});
-  const developer = developers.find((dev) => compareSync(apiKey, dev.apiKey));
-  console.timeEnd('Finding developer account');
-  return developer;
-};
+import {
+  handleRequest,
+  FALLBACK_API_KEY,
+  checkDeveloperAPIKey,
+  findDeveloper,
+} from '../controllers/developers/utils';
 
 export default async (req, res, next) => {
   try {
-    const { apiLimit } = req.query;
-    let apiKey = req.headers['X-API-Key'] || req.headers['x-api-key'];
+    let { apiToken: apiKey, apiLimit } = handleRequest(req);
+    apiLimit = 2500;
```

Comment:
why are you setting the apiLimit again? i don't think you need to explore `apiLimit` from `handleRequest` since `PROD_LIMIT` is the constant that determines the limit for API usage